homeassistant/components/electric_kiwi/sensor.py

```diff
@@ -0,0 +1,75 @@
+"""Support for Electric Kiwi account balance."""
+from datetime import timedelta
+import logging
+
+import async_timeout
+
+from homeassistant.components.sensor import SensorEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    ATTR_ATTRIBUTION,
+    ATTR_FRIENDLY_NAME,
+    CURRENCY_DOLLAR
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from electrickiwi_api import ElectricKiwiApi
+from electrickiwi_api.model import AccountBalance
+
+from .const import DOMAIN
+from ...util import Throttle
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = timedelta(hours=24)
+
+ATTRIBUTION = "Data provided by Electric Kiwi Juice Hacker API"
+FRIENDLY_NAME = "Electric Kiwi Account Balance"
+
+
+async def async_setup_entry(
+        hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Electric Kiwi Sensor Setup."""
+    api: ElectricKiwiApi = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities([ElectricKiwiBalanceSensor(api)], True)
+
+
+class ElectricKiwiBalanceSensor(SensorEntity):
```

Comment:
I would recommend adding a `unique_id` to this sensor, so it can be renamed etc in the UI.

Could probably use account number for that.

Reply:
Can you point me to the documentation for this? Yes account number would be good but I can only retrieve that after setup, so not sure how easy that is.

Reply:
nevermind sorted it out